artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/AbstractJournalStorageManager.java

```diff
@@ -1839,6 +1838,9 @@ private void loadSinglePreparedTransaction(PostOffice postOffice,
             }
             case JournalRecordIds.ADD_MESSAGE_PROTOCOL: {
                Message message = decodeMessage(pools, buff);
+               if (storedLargeMessages != null && message.isLargeMessage() && storedLargeMessages.remove(record.id)) {
+                  logger.debug("PreparedTX load removing stored large message {}", record.id);
```

Comment:
might be useful to differentiate the message from the currently-identical log message above to help discern which occurred without resorting to line-number output comparisons (Aside: I'm just assuming there is some 'known'/'obvious' reason for the 2 different record types doing such similar thing to extent of having such simialr log messages).

also, guessing same about gate.